src/types/index.ts

```diff
@@ -0,0 +1,47 @@
+interface DataProps { 
+    albumId: number;
+    id: number,
+    thumbnailUrl: string;
+    title: string;
+    url: string;
+}
+
+export interface ButtonProps {
+    value: number;
+    key:number;
+    chooseLimit: (value:number) => void;
+    fetchBlogs: (page:number, limit:number) => DataProps[] | object;
+    changeCurrentPage: () => void;
+};
+
+
+export interface HeaderProps {
+    limit: number;
+    chooseLimit: (value:number) => void;
+    fetchBlogs: (page:number, limit:number) => DataProps[] | object;
+    changeCurrentPage: () => void;
+}
+
+export interface Blog {
+    id: number,
+    title: string,
+    url:string,
+}
+
+export interface ButtonsProps {
```

Comment:
Just one update. Since you should be able to pass all html `<button>` props to the your component `Button`, extend this interface from `React.HTMLProps<HTMLButtonElement>`

Reply:
And since you change `value` type and have a bit custom `onChange`, you need exclude them from html button props

Reply:
Use TS `Exclude<React.HTMLProps<HTMLButtonElement>, 'value' | 'onChange'>`

Reply:
Always do that way when you implementing buttons, inputs, checkboxes and other UI components that have HTML little brothers inside which we pass props to)